frontend/src/metabase/reference/components/Formula.tsx

```diff
@@ -0,0 +1,52 @@
+import cx from "classnames";
+import { TransitionGroup, CSSTransition } from "react-transition-group";
+import { t } from "ttag";
+
+import { QueryDefinition } from "metabase/admin/datamodel/components/QueryDefinition";
+import { Icon } from "metabase/ui";
+import type { Metric, Segment } from "metabase-types/api";
+
+import S from "./Formula.module.css";
+
+export const Formula = ({
+  type,
+  entity,
+  isExpanded,
+  expandFormula,
+  collapseFormula,
+}: {
+  type: "segment" | "metric";
```

Comment:
I wonder if we should make this so that you need to pass a Segment when you pass type=segment etc, but it's only used in js files for now so it won't be easy to verify if that's the case

Reply:
Ah I see what you mean. I think it's worth changing for now, those components will eventually get converted